docs/output_file_model_documentation.md

```diff
@@ -0,0 +1,818 @@
+
+- **Use `OutputFile`** when:
+  - Returning data to API users (REST/GraphQL responses)
+  - Accepting user input and need Pydantic validation
+  - Serializing to JSON for external consumption
+
+- **Use `OutputFileInternal`** when:
+  - Querying the database
+  - Performing GCS validation and metadata extraction
+  - Implementing business logic that doesn't directly interface with users
+  - Need access to database-specific methods like `from_db()`
+
+**Example conversion:**
+
+```python
+# API receives Pydantic model, converts to internal for DB operations
+external_model = OutputFile(path="gs://bucket/file.vcf", ...)
+internal_model = external_model.to_internal()
+
+# Database returns internal model, converts to external for API response
+db_row = await fetch_from_db(...)
+internal_model = OutputFileInternal.from_db(db_row)
+external_model = internal_model.to_external()
+```
+
+### Data Flow Overview
+
+The following diagram shows how OutputFile data flows through the system when you create an analysis with outputs:
+
+```mermaid
+graph TD
+    A[User submits analysis with outputs] --> B{Parse output structure}
+    B --> C[Extract GCS paths]
+    C --> D[Validate files in GCS]
+    D --> E{File exists?}
+    E -->|Yes| F[Extract metadata<br/>checksum, size]
+    E -->|No| G[Mark as invalid<br/>store as string]
+    F --> H[Create OutputFile record]
+    G --> I[Store raw output string]
+    H --> J[Link to Analysis via<br/>analysis_outputs table]
+    I --> J
+    J --> K[Return enriched response<br/>to user]
+
+    style F fill:#90EE90
+    style G fill:#FFB6C1
+    style K fill:#87CEEB
+```
+
+### File Validation
+
+Every OutputFile is automatically validated against Google Cloud Storage when created or updated.
+
+**Validation Process:**
+
+1. **Existence Check**: Metamist attempts to access the file at the specified GCS path
+2. **Metadata Extraction**: If the file exists, GCS metadata is retrieved:
+   - **CRC32C Checksum**: For data integrity verification
+   - **File Size**: In bytes
+3. **Validity Flag**: The `valid` field is set to `true` if the file is accessible, `false` otherwise
+
+**Validation Results:**
+
+| Validation Status | Response Format | Use Case |
+|------------------|-----------------|----------|
+| ✅ File exists and accessible | Full OutputFile object with `valid: true` | Normal operation |
+| ❌ File not found or inaccessible | Simple string (no OutputFile created) | Planned outputs, missing files, permission issues |
+
+> [!NOTE]
+>
+> **All OutputFiles have `valid: true`**: Because the code returns `None` for any file that fails validation (see `create_or_update_output_file()`), all OutputFile records in the database have `valid: true`. Files that can't be validated simply don't get OutputFile records.
+>
+> [!CAUTION]
+> **Matrix Table (.mt) files** are currently not supported. These Hail-specific files are treated as directories in GCS and cannot be validated using the standard file validation process.
```

Comment:
Would be good to be clear here what "not supported" means. Is the file marked as invalid?